### PR TITLE
python.pkgs.{tensorflow-probability,dm-sonnet,graph_nets}: init

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

I want to work with graph networks in tensorflow. The canonical library depends on `dm-sonnet`, which in turn depends on `tensorflow-probability`.

The `dm-sonnet` build is based on the `tensorflow` build (though greatly simplified), since I couldn't find any documentation for bazel builds.

I tested this by building a model to predict the shortest path on graphs, as in the [example](https://colab.research.google.com/github/deepmind/graph_nets/blob/master/graph_nets/demos/shortest_path.ipynb).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by timokau** [comment]
@GrahamcOfBorg build python2.pkgs.graph_nets python3.pkgs.graph_nets

CC @jyp who contributed to the tensorflow package a couple of times and may or may not be interested in reviewing this :)

**Comment by timokau** [comment]
ofBorg failure:
> The TensorFlow library was compiled to use AVX instructions, but these aren't available on your machine.
Seems to be an issue with the binary build of tensorflow and an older ofBorg builder. Not related to this.

**Review by FRidh — APPROVED** [review]

File: pkgs/development/python-modules/tensorflow-probability/default.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  # Tests have an invalid import (`tensorflow_probability.opensource`), should
+  # be resolved in the next version with
+  # https://github.com/tensorflow/probability/commit/77d5957f2f0bdddcb46582799cd9c5c5167a1a40
+  doCheck = false;
+  checkPhase = ''
+    py.test
+  '';
+
+  meta = with lib; {
+    description = "a library for probabilistic reasoning and statistical analysis";
```
Comment (markuskowa) [inline pkgs/development/python-modules/tensorflow-probability/default.nix]:
```suggestion
    description = "Library for probabilistic reasoning and statistical analysis";
```

Comment (timokau) [reply pkgs/development/python-modules/tensorflow-probability/default.nix]: Applied in rebase, thanks!

**Pull request merged**